roles/artifacts/tasks/edpm.yml

```diff
@@ -42,11 +42,21 @@
             - cifmw_edpm_deploy_extra_vars is defined
           ansible.builtin.set_fact:
             ssh_key_file: "{{ cifmw_edpm_deploy_extra_vars.SSH_KEY_FILE }}"
```

Comment:
ok so i am seeing this error

fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'dict object' has no attribute 'SSH_KEY_FILE'. 'dict object' has no attribute 'SSH_KEY_FILE'\n\nThe error appears to be in '/home/zuul/src/github.com/openstack-k8s-operators/ci-framework/roles/artifacts/tasks/edpm.yml': line 39, column 11, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n        - name: Extract Compute from zuul mapping if any\n          ^ here\n"}

so that means cifmw_edpm_deploy_extra_vars does not have SSH_KEY_FILE defined for standalone host.
ssh_key_file: "{{ cifmw_edpm_deploy_extra_vars.SSH_KEY_FILE }}"

Reply:
i wonder if we can just use ssh_key_file: "{{ cifmw_artifacts_basedir }}/artifacts/edpm/ansibleee-ssh-key-id_rsa"

or 
SSH_KEY_FILE: "{{ ansible_user_dir }}/.ssh/id_cifw"

https://github.com/openstack-k8s-operators/ci-framework/blob/64d1de7421b9e1d0211862bb6d9f2789ef288e0b/hooks/playbooks/fetch_compute_facts.yml#L105

looking at https://review.rdoproject.org/zuul/job/data-plane-adoption-osp-17-to-extracted-crc

we could use standalone_private_key in the adoptin case sepcificly but it woudl probaly be better to define cifmw_edpm_deploy_extra_vars.SSH_KEY_FILE as {{standalone_private_key}} instead to make it work as expected.

Reply:
ok this is normlaly set here

https://github.com/openstack-k8s-operators/ci-framework/blob/64d1de7421b9e1d0211862bb6d9f2789ef288e0b/hooks/playbooks/fetch_compute_facts.yml#L105

in edpm_install_yamls_vars and then that get added cifmw_edpm_deploy_extra_vars
here
https://github.com/openstack-k8s-operators/ci-framework/blob/64d1de7421b9e1d0211862bb6d9f2789ef288e0b/hooks/playbooks/fetch_compute_facts.yml#L116-L117

so if that is not being set computes_len is not > 0 

and since that defined here

https://github.com/openstack-k8s-operators/ci-framework/blob/64d1de7421b9e1d0211862bb6d9f2789ef288e0b/hooks/playbooks/fetch_compute_facts.yml#L33-L35

it means the standalone node is not part of the compute group in the inventory

that is being exectued as a hook so its not the zuul inventory its the inventory used by the ansible-playbook command in that is runing hte hook.

Reply:
the groups by defualt come from the nodeset and we can see in this case the nodeset does not define the standalone node to be part of the compute group 

https://github.com/openstack-k8s-operators/ci-framework/blob/64d1de7421b9e1d0211862bb6d9f2789ef288e0b/zuul.d/nodeset.yaml#L46-L63


so if i fix that it should resolve that issue and the key shoudl be set properly.